src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -603,6 +674,15 @@ private unsafe int HandleEventPeerStreamStarted(ref PEER_STREAM_STARTED_DATA dat
         data.Flags |= QUIC_STREAM_OPEN_FLAGS.DELAY_ID_FC_UPDATES;
         return QUIC_STATUS_SUCCESS;
     }
+    private unsafe int HandleEventStreamsAvailable(ref STREAMS_AVAILABLE_DATA data)
```

Comment:
I like this idea. I can try that. It would also resolve the issue with the `_decrementAvailableStreamCount` callback. However, I'll not push on adding this to MsQuic, until we have the API approved, we might end up with MAX_STREAM_ID or different value being reported by the callback...

Reply:
Draft change in https://github.com/dotnet/runtime/commit/36210649c502c006e0bdd1439f9ba4c8a976df28
MsQuic change in https://github.com/microsoft/msquic/commit/d47f9c189a0070abe791e23daac9fd4f57c6f6cb

Reply:
As said above, I'm keeping the changes as they are now. I do plan to push for the change in MsQuic, however that will take time and also will only be part of the new release from their main (i.e. 2.4) at best. And I don't want to keep this PR open and waiting for that.

Reply:
Makes sense. I'd like to see it happen but it's not worth blocking changes here.